apps/freeablo/engine/engineinputmanager.cpp

```diff
@@ -123,31 +123,35 @@ namespace Engine
     {
         NuklearMisc::handleNuklearMouseEvent(mNkCtx, x, y, key, true, isDoubleClick);
 
-        if (key == Input::Key::KEY_LEFT_MOUSE)
-        {
-            mMousePosition = Misc::Point{x, y};
-            mMouseDown = true;
-            mClick = true;
-        }
-        if (key == Input::Key::KEY_RIGHT_MOUSE)
+        mMousePosition = Misc::Point{x, y};
+
+        // Only pass relevant mouse clicks to game engine.
+        if (!mPaused && !nk_item_is_any_active(mNkCtx) && !mGuiManager->isModalDlgShown())
```

Comment:
`mMouseDown`/`mRightMouseDown` etc are now only relevant for the game engine.
The other option was to add additional state e.g. `mMouseDownIgnore`/`mRightMouseDownIgnore` which seemed worse.
Seems to work well, let me know if I've missed/broken anything